packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/scoped_cluster_client.ts

```diff
@@ -10,8 +10,39 @@ import type { ElasticsearchClient, IScopedClusterClient } from '@kbn/core-elasti
 
 /** @internal **/
 export class ScopedClusterClient implements IScopedClusterClient {
-  constructor(
-    public readonly asInternalUser: ElasticsearchClient,
-    public readonly asCurrentUser: ElasticsearchClient
-  ) {}
+  public readonly asInternalUser;
+
+  readonly #asCurrentUserClientFactory: () => ElasticsearchClient;
+  readonly #asSecondaryAuthClientFactory: () => ElasticsearchClient;
+
+  #asCurrentUserClient?: ElasticsearchClient;
+  #asSecondaryAuthClient?: ElasticsearchClient;
+
+  constructor({
```

Comment:
It seemed so much simpler before lazily creating clients!

Reply:
Yeah, it sure was. But I remember that the time spent in client creation is not negligible, so unfortunatly we had to optimize that thing, and usually perf optimization is at the expense of code simplicity.